// @mew/request/test/weixin/weixin.test.ts
import axios, {AxiosInstance} from "axios";
import {DingTalk, Mc} from "../../src/main";

export interface AuthProperty {
    username: string;
    password: string;
}

//auth
export interface AuthData {
    access_token: string
    expiration: number
    overdueDateTs: number
    overdueDate: string
}

const service = axios.create({
    baseURL: 'https://api.mew-iot.cn',
    // adapter: axiosAdapterUniapp as AxiosAdapter,
    timeout: 10000
})

const mc = Mc(service)

const appId = 'wx542e777807090c4e' // 喜利得小程序

/**
 * 提供基础数据
 */
describe('weixin',()=>{
    const mc = Mc(service)
    let access_token = ''
    beforeAll(async()=>{
        // // get app info
        // const res = await mc.getAccessToken(appId)
        // console.log(18, res)
        // access_token = res.data.accessToken
    })
    const username = 'test'
    const workorder = new Date().getTime().toString()

    test('获取微信用户信息', async () => {
        const code ="02103K000wZIxN1TV1000gmfzF403K0j"
        const res = await mc.getAppUserInfo({appId,code})
        // console.log(15, res)
        expect(res.code).toBe(0)
        expect(res.msg).toBe('success')
        expect(res.data.session_key).not.toBe('')
        expect(res.data.openid).not.toBe('')
        expect(res.data.unionid).not.toBe('')
    }, 3000)

    test('发送微信模板消息-上架通知', async () => {
        const data = {
                "character_string1": {
                "value": "339208499"
            },
                "thing2": {
                "value": "广州市海珠区新港中路397号"
            }
        }
        const json ={
            appId,
            template_id: 'JhO4leaUMd_TeECaM2-n1iFZ8RGSKK9Ww7hA5YJBlqQ',
            touser: 'o3SqR5eLnxS5HRIm83KKShTzbSWg',
            page: 'index',
            form_id: 'FORMID',
            emphasis_keyword: 'keyword1.DATA',
            data: data
        }
        const res = await mc.sendWeiXinMessage(json)
        expect(res.code).toBe(0)
        expect(res.msg).toBe('success')
        expect(res.data.errcode).toBe(0)
        expect(res.data.errmsg).toBe('ok')
    }, 2000)

    test('发送微信模板消息-设备补货提醒', async () => {
        const json ={
            appId,
            template_id: '5bCe8RFNbhFWT0NXVIIvCcM0InWVmQ9OT5DmE_grN3Y',
            touser: 'o3SqR5eLnxS5HRIm83KKShTzbSWg',
            page: 'index',
            form_id: 'FORMID',
            emphasis_keyword: 'keyword1.DATA',
            data: {
                "character_string3": {"value": "86uwe"},
                "thing4": {"value": "喜利得租赁柜"},
                "time5": {"value": "2020-02-11 15:48:10"}
            }
        }
        // console.log(json)
        const res = await mc.sendWeiXinMessage(json)
        // console.log(100, res)
        expect(res.code).toBe(0)
        expect(res.msg).toBe('success')
        expect(res.data.errcode).toBe(0)
        expect(res.data.errmsg).toBe('ok')
    }, 2000)

    test('发送微信模板消息-付款成功通知', async () => {
        const json ={
            appId,
            template_id: 'OyB7mOGSF4J6k7FyPwjkWZJYiJXTFKVqUvu0JUTwWXY',
            touser: 'o3SqR5eLnxS5HRIm83KKShTzbSWg',
            page: 'index',
            form_id: 'FORMID',
            emphasis_keyword: 'keyword1.DATA',
            data: {
                "amount1": {"value": "200元"},
                "character_string2": {"value": "123456789"},
                "thing3": {"value": "提货码：12345678"}
            }
        }
        console.log(json)
        const res = await mc.sendWeiXinMessage(json)
        expect(res.code).toBe(0)
        expect(res.msg).toBe('success')
        expect(res.data.errcode).toBe(0)
        expect(res.data.errmsg).toBe('ok')
    }, 2000)

    test('发送微信模板消息-领用归还提醒', async () => {
        const json ={
            appId,
            template_id: 'Q4-fV8wEFj0ePgcL2fSNvNPCMURw5oItj8PCRlB0ob8',
            touser: 'o3SqR5eLnxS5HRIm83KKShTzbSWg',
            page: 'index',
            form_id: 'FORMID',
            emphasis_keyword: 'keyword1.DATA',
            data: {
                "thing1": {"value": "领用"},
                "thing2": {"value": "李亮"},
                "thing3": {"value": "1"},
                "thing4": {"value": "领用归还"},
                "time5": {"value": "2022年03月30日 15:00"}
            }
        }
        console.log(json)
        const res = await mc.sendWeiXinMessage(json)
        console.log(res)
        expect(res.code).toBe(0)
        expect(res.msg).toBe('success')
        expect(res.data.errcode).toBe(0)
        expect(res.data.errmsg).toBe('ok')
    }, 3000)

    test('发送微信模板消息-领用归还提醒', async () => {
        const json ={
            appId,
            template_id: 'Q4-fV8wEFj0ePgcL2fSNvNPCMURw5oItj8PCRlB0ob8',
            touser: 'o3SqR5eLnxS5HRIm83KKShTzbSWg',
            page: 'index',
            form_id: 'FORMID',
            emphasis_keyword: 'keyword1.DATA',
            data: {
                "thing1": {"value": "领用"},
                "thing2": {"value": "李亮"},
                "thing3": {"value": "1"},
                "thing4": {"value": "领用归还"},
                "time5": {"value": "2022年03月30日 15:00"}
            }
        }
        console.log(json)
        const res = await mc.sendWeiXinMessage(json)
        console.log(res)
        expect(res.code).toBe(0)
        expect(res.msg).toBe('success')
        expect(res.data.errcode).toBe(0)
        expect(res.data.errmsg).toBe('ok')
    }, 3000)

    test('发送微信模板消息-库存不足提醒', async () => {
        const json ={
            appId,
            template_id: 'ylwJJ5juy6nH6nQVi-KbVdaPq0isXaXEKdXeB7hhl2g',
            touser: 'o3SqR5eLnxS5HRIm83KKShTzbSWg',
            page: 'index',
            form_id: 'FORMID',
            emphasis_keyword: 'keyword1.DATA',
            data: {
                "number1": {"value": "182106003"},
                "thing2": {"value": "上海市嘉定区"},
                "number3": {"value": 1234},
                "thing4": {"value": "TX-钻头"},
                "number5": {"value": "1"}
            }
        }
        console.log(json)
        const res = await mc.sendWeiXinMessage(json)
        console.log(res)
        expect(res.code).toBe(0)
        expect(res.msg).toBe('success')
        expect(res.data.errcode).toBe(0)
        expect(res.data.errmsg).toBe('ok')
    }, 3000)

    test('发送微信模板消息-设备状态通知', async () => {
        const json ={
            appId,
            template_id: 'cljeY4GrAZcCHVd8-5VgDOvl1v2LE0yBwPAAsgxNrcE',
            touser: 'o3SqR5eLnxS5HRIm83KKShTzbSWg',
            page: 'index',
            form_id: 'FORMID',
            emphasis_keyword: 'keyword1.DATA',
            data: {
                "thing1": {"value": "喜利得售卖柜"},
                "thing2": {"value": "上海市嘉定区"},
                "phrase3": {"value": "设备已开机"}
            }
        }
        console.log(json)
        const res = await mc.sendWeiXinMessage(json)
        console.log(res)
        expect(res.code).toBe(0)
        expect(res.msg).toBe('success')
        expect(res.data.errcode).toBe(0)
        expect(res.data.errmsg).toBe('ok')
    }, 3000)
})
