// @mew/request/dist/dingtalk/dingtalk.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DingTalk = void 0;
var tslib_1 = require("tslib");
var request_1 = require("../request");
var main_1 = require("../main");
// const appId = '5000000002334316' // 钉钉企业第三方应用-智能配发柜
var DingTalk = /** @class */ (function () {
    function DingTalk(appId, service, isDebugMode) {
        if (isDebugMode === void 0) { isDebugMode = false; }
        this._access_token = '';
        this._appId = appId;
        this._service = service;
        this._isDebugMode = isDebugMode;
        this.init();
    }
    // 初始化
    DingTalk.prototype.init = function () {
        return tslib_1.__awaiter(this, void 0, void 0, function () {
            var mc, res;
            return tslib_1.__generator(this, function (_a) {
                switch (_a.label) {
                    case 0:
                        mc = main_1.Mc(this._service, this._isDebugMode);
                        return [4 /*yield*/, mc.getAccessToken(this._appId)
                            // console.log(71, res)
                        ];
                    case 1:
                        res = _a.sent();
                        // console.log(71, res)
                        this._access_token = res.data.accessToken;
                        return [2 /*return*/];
                }
            });
        });
    };
    DingTalk.prototype.getMsAccessToken = function () {
        return tslib_1.__awaiter(this, void 0, void 0, function () {
            var json, res;
            return tslib_1.__generator(this, function (_a) {
                switch (_a.label) {
                    case 0:
                        json = { username: 'admin', password: '111111' };
                        return [4 /*yield*/, request_1.mewRequest('/auth', json, 'POST', this._service, this._isDebugMode)];
                    case 1:
                        res = _a.sent();
                        return [2 /*return*/, res.access_token];
                }
            });
        });
    };
    DingTalk.prototype.getDingTalkUserinfo = function (code) {
        return tslib_1.__awaiter(this, void 0, void 0, function () {
            var mc;
            return tslib_1.__generator(this, function (_a) {
                switch (_a.label) {
                    case 0:
                        mc = main_1.Mc(this._service, this._isDebugMode);
                        return [4 /*yield*/, mc.getDingTalkUserInfo({ appId: this._appId, code: code })];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    /**
     * 备品上架通知
     */
    DingTalk.prototype.sendTemplateMsgOnShelf = function (userid_list, data) {
        return tslib_1.__awaiter(this, void 0, void 0, function () {
            var mc, json;
            return tslib_1.__generator(this, function (_a) {
                switch (_a.label) {
                    case 0:
                        mc = main_1.Mc(this._service, this._isDebugMode);
                        json = {
                            appId: this._appId,
                            template_id: 'f63544a6c20c4069b6833cd9be3fd108',
                            data: JSON.stringify(data),
                            userid_list: userid_list
                        };
                        return [4 /*yield*/, mc.sendDingTalkMessage(json)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    /**
     * 备品预检通知
     */
    DingTalk.prototype.sendTemplateMsgPreInspection = function (userid_list, data) {
        return tslib_1.__awaiter(this, void 0, void 0, function () {
            var mc, json;
            return tslib_1.__generator(this, function (_a) {
                switch (_a.label) {
                    case 0:
                        mc = main_1.Mc(this._service, this._isDebugMode);
                        json = {
                            appId: this._appId,
                            template_id: '559d93d3f455459895a7b81eda600792',
                            data: JSON.stringify(data),
                            userid_list: userid_list
                        };
                        return [4 /*yield*/, mc.sendDingTalkMessage(json)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    /**
     * 备品换货通知
     */
    DingTalk.prototype.sendTemplateMsgChange = function (userid_list, data) {
        return tslib_1.__awaiter(this, void 0, void 0, function () {
            var mc, json;
            return tslib_1.__generator(this, function (_a) {
                switch (_a.label) {
                    case 0:
                        mc = main_1.Mc(this._service);
                        json = {
                            appId: this._appId,
                            template_id: '29e7205d2c284c1cb4e72fa45e1af77d',
                            data: JSON.stringify(data),
                            userid_list: userid_list
                        };
                        return [4 /*yield*/, mc.sendDingTalkMessage(json)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    /**
     * 备品领用通知
     */
    DingTalk.prototype.sendTemplateMsgPickUp = function (userid_list, data) {
        return tslib_1.__awaiter(this, void 0, void 0, function () {
            var mc, json;
            return tslib_1.__generator(this, function (_a) {
                switch (_a.label) {
                    case 0:
                        mc = main_1.Mc(this._service, this._isDebugMode);
                        json = {
                            appId: this._appId,
                            template_id: '257a3f214b244e91a4656700eb356fa2',
                            data: JSON.stringify(data),
                            userid_list: userid_list
                        };
                        return [4 /*yield*/, mc.sendDingTalkMessage(json)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    /**
     * 备品退销通知
     */
    DingTalk.prototype.sendTemplateMsgReturn = function (userid_list, data) {
        return tslib_1.__awaiter(this, void 0, void 0, function () {
            var mc, json;
            return tslib_1.__generator(this, function (_a) {
                switch (_a.label) {
                    case 0:
                        mc = main_1.Mc(this._service, this._isDebugMode);
                        json = {
                            appId: this._appId,
                            template_id: '6e7d45c4c1974053a7c1ae23df013277',
                            data: JSON.stringify(data),
                            userid_list: userid_list
                        };
                        return [4 /*yield*/, mc.sendDingTalkMessage(json)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    /**
     * 智能配发柜-备品领用委托通知
     */
    DingTalk.prototype.sendTemplateMsgPickUpEntrust = function () {
        return tslib_1.__awaiter(this, void 0, void 0, function () {
            return tslib_1.__generator(this, function (_a) {
                return [2 /*return*/];
            });
        });
    };
    return DingTalk;
}());
exports.DingTalk = DingTalk;
