// @mew/request/src/ms/dashboard.ts
import {IRes, IResType, mewMsPost, mewRequest} from "../request";
import axios, {AxiosAdapter, AxiosInstance, AxiosRequestConfig, Method} from "axios";

export interface FindSqlProperty {
    cid: number
    dateFrom: number
    dateTo: number
}

export interface DepartmentFrequencyProperty {
    id: number
    count: string
    region: string
    rate: string
}

export interface LogByDay {
    day: string
    in: number
    out: number
}

export interface TopTenUsageTimes {
    toolId: number
    deviceId: string
    partDes: string
    number: string
}

export interface TopTenUsers {
    username: string
    count: string
}

export interface DeviceStatus {
    online: number
    total: number
}

export interface BmProperty {
    companyId: number;
    deviceId: string;
    startTime?: number;
    endTime?: number;
    type:number;
}

export function Dashboard(service: any, isDebugMode: boolean = false) {
    return {
        // function
        cacheGetData: async () => {
            return await mewMsPost('/cache/get', {key: "localStatus", index: 0}, service, isDebugMode)
        },

        /**
         * 获取工具借还情况,根据天
         */
        toolSituation: async function (json: any): Promise<IResType<LogByDay>> {
            return await mewMsPost('/statistics/logByDay', json, service, isDebugMode)
        },

        /**
         * 获取工具使用前十
         * 工具使用排行榜top10
         * /log/getTopTenUsageTimes
         */
        toolTopTen: async function (json: any): Promise<IResType<TopTenUsageTimes>> {
            return await mewMsPost('/statistics/topTenUsageTimes', json, service, isDebugMode)
        },

        /**
         * 获取工具使用后十
         * 工具使用排行榜top10
         * /log/getTopTenUsageTimes
         */
        toolBottomTen: async function (json: any): Promise<IResType<TopTenUsageTimes>> {
            return await mewMsPost('/statistics/bottomTenUsageTimes', json, service, isDebugMode)
        },

        /**
         * 部门使用情况
         */
        departmentFrequency: async function (json: any): Promise<IResType<DepartmentFrequencyProperty>> {
            return await mewMsPost('/statistics/departmentFrequency', json, service, isDebugMode)
        },

        /**
         * 班组使用情况
         */
        teamFrequency: async function (json: any): Promise<IResType<DepartmentFrequencyProperty>> {
            return await mewMsPost('/statistics/departmentFrequency', json, service, isDebugMode)
        },

        /**
         * 班组使用情况
         */
        topTenUsers: async function (json: any): Promise<IResType<TopTenUsers>> {
            return await mewMsPost('/statistics/topTenUsers', json, service, isDebugMode)
        },

        /**
         * 设备状态
         */
        deviceStatus: async function (json: any): Promise<IRes<DeviceStatus>> {
            return await mewRequest('/statistics/deviceStatus', json, 'POST', service, isDebugMode)
        },

        /**
         * 大屏数据
         */
        bmData: async function (json: BmProperty): Promise<IRes> {
            return await mewRequest('/statistics/bmData', json, 'POST', service, isDebugMode)
        },

        /**
         * 大屏数据
         */
        lackToolList: async function (json: BmProperty): Promise<IRes> {
            return await mewRequest('/statistics/lackToolList', json, 'POST', service, isDebugMode)
        }
    }
}
