// @mew/request/dist/main.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.mewRequest = exports.mewMsPost = exports.DingTalk = exports.RabbitApi = exports.Rabbit = exports.Dashboard = exports.Common = exports.Uc = exports.Tc = exports.Mc = exports.Lc = exports.Gc = exports.Dc = exports.Ac = exports.getAuth = exports.AxiosEnv = void 0;
var tslib_1 = require("tslib");
var ac_1 = require("./ms/ac");
Object.defineProperty(exports, "Ac", { enumerable: true, get: function () { return ac_1.Ac; } });
var dc_1 = require("./ms/dc");
Object.defineProperty(exports, "Dc", { enumerable: true, get: function () { return dc_1.Dc; } });
var gc_1 = require("./ms/gc");
Object.defineProperty(exports, "Gc", { enumerable: true, get: function () { return gc_1.Gc; } });
var lc_1 = require("./ms/lc");
Object.defineProperty(exports, "Lc", { enumerable: true, get: function () { return lc_1.Lc; } });
var mc_1 = require("./ms/mc");
Object.defineProperty(exports, "getAuth", { enumerable: true, get: function () { return mc_1.getAuth; } });
Object.defineProperty(exports, "Mc", { enumerable: true, get: function () { return mc_1.Mc; } });
var tc_1 = require("./ms/tc");
Object.defineProperty(exports, "Tc", { enumerable: true, get: function () { return tc_1.Tc; } });
var uc_1 = require("./ms/uc");
Object.defineProperty(exports, "Uc", { enumerable: true, get: function () { return uc_1.Uc; } });
var rabbit_1 = require("./ms/rabbit");
Object.defineProperty(exports, "Rabbit", { enumerable: true, get: function () { return rabbit_1.Rabbit; } });
var common_1 = require("./ms/common");
Object.defineProperty(exports, "Common", { enumerable: true, get: function () { return common_1.Common; } });
var dashboard_1 = require("./ms/dashboard");
Object.defineProperty(exports, "Dashboard", { enumerable: true, get: function () { return dashboard_1.Dashboard; } });
var rabbit_api_1 = tslib_1.__importDefault(require("./rabbit.api"));
exports.RabbitApi = rabbit_api_1.default;
var dingtalk_1 = require("./dingtalk/dingtalk");
Object.defineProperty(exports, "DingTalk", { enumerable: true, get: function () { return dingtalk_1.DingTalk; } });
var request_1 = require("./request");
Object.defineProperty(exports, "mewMsPost", { enumerable: true, get: function () { return request_1.mewMsPost; } });
Object.defineProperty(exports, "mewRequest", { enumerable: true, get: function () { return request_1.mewRequest; } });
var AxiosEnv;
(function (AxiosEnv) {
    AxiosEnv[AxiosEnv["default"] = 0] = "default";
    AxiosEnv[AxiosEnv["uniapp"] = 1] = "uniapp";
})(AxiosEnv = exports.AxiosEnv || (exports.AxiosEnv = {}));
