// @mew/request/src/weixin/weixin.ts
import {AxiosInstance} from "axios";
import {AppSendDingTalkMessage, AppSendWxMessage, Mc} from "../ms/mc";
import {mewRequest} from "../request";


export class WeiXin {
    _appId: string
    _service: AxiosInstance
    _access_token: string = ''
    constructor(appId: string, service: AxiosInstance) {
        this._appId = appId
        this._service = service
        this.init()
    }
    private async init(){
        const mc = Mc(this._service)
        // get app info
        const res = await mc.getAccessToken(this._appId)
        console.log(18, res)
        this._access_token = res.data.accessToken
    }

    private async getMsAccessToken() {
        const json = {username: 'admin', password: '111111'}
        const res = await mewRequest('/auth', json, 'POST', this._service)
        return res.access_token
    }

    /**
     * 仅支持在微信小程序内调用才有效
     */
    async getUserInfo(code:string){
        const mc = Mc(this._service)
        return await mc.getAppUserInfo({appId:this._appId,code})
    }

    async sendTemplateMessage(touser,data){
        const mc = Mc(this._service)
        const json:AppSendWxMessage = {
            appId: this._appId,
            template_id: 'JhO4leaUMd_TeECaM2-n1iFZ8RGSKK9Ww7hA5YJBlqQ',
            touser,
            page: 'index',
            form_id: 'FORMID',
            emphasis_keyword: 'keyword1.DATA',
            data
        }
        return await mc.sendWeiXinMessage(json)
    }
    // // 设备补货提醒
    // async sendTemplateMessage(touser,data){
    //     const mc = Mc(this._service)
    //     const json:AppSendWxMessage = {
    //         appId: this._appId,
    //         template_id: 'JhO4leaUMd_TeECaM2-n1iFZ8RGSKK9Ww7hA5YJBlqQ',
    //         touser,
    //         page: 'index',
    //         form_id: 'FORMID',
    //         emphasis_keyword: 'keyword1.DATA',
    //         data
    //     }
    //     return await mc.sendWeiXinMessage(json)
    // }
}
