// @mew/request/dist/ms/ac.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Ac = void 0;
var tslib_1 = require("tslib");
var request_1 = require("../request");
function Ac(service, isDebugMode) {
    var _this = this;
    if (isDebugMode === void 0) { isDebugMode = false; }
    return {
        process: new request_1.Request('/ac/process', service, isDebugMode),
        process_type: new request_1.Request('/ac/process_type', service, isDebugMode),
        file: new request_1.Request('/ac/file', service, isDebugMode),
        setting: new request_1.Request('/ac/setting', service, isDebugMode),
        carousel: new request_1.Request('/ac/carousel', service, isDebugMode),
        // class
        task: new request_1.Request('/ac/task', service, isDebugMode),
        alarm: new request_1.Request('/ac/alarm', service, isDebugMode),
        feedback: new request_1.Request('/ac/feedback', service, isDebugMode),
        // function
        getVersion: function () { return request_1.mewRequest('/ac/select/version', {}, 'POST', service); },
        processNext: function (json) { return request_1.mewRequest('/ac/process/next', json, 'POST', service, isDebugMode); },
        processStart: function (json) { return request_1.mewRequest('/ac/process/start', json, 'POST', service, isDebugMode); },
        messageNotice: function (json) { return request_1.mewRequest('/ac/message/message_notice', json, 'POST', service, isDebugMode); },
        getTaskNumber: function (json) { return tslib_1.__awaiter(_this, void 0, void 0, function () {
            var res;
            return tslib_1.__generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, request_1.mewRequest('/ac/task/get_task_number', json, 'POST', service, isDebugMode)];
                    case 1:
                        res = _a.sent();
                        return [2 /*return*/, {
                                code: res.code,
                                data: res.data
                            }];
                }
            });
        }); }
    };
}
exports.Ac = Ac;
