// @mew/request/dist/ms/common.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Common = void 0;
var tslib_1 = require("tslib");
var request_1 = require("../request");
function Common(service, isDebugMode) {
    var _this = this;
    if (isDebugMode === void 0) { isDebugMode = false; }
    return {
        // function
        cacheGetData: function (service) { return function () { return tslib_1.__awaiter(_this, void 0, void 0, function () {
            return tslib_1.__generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, request_1.mewMsPost('/cache/get', { key: "localStatus", index: 0 }, service, isDebugMode)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        }); }; },
        /**
         * 获取缓存
         * {key: "conf"}
         */
        cacheGet: function (service) { return function () { return tslib_1.__awaiter(_this, void 0, void 0, function () {
            return tslib_1.__generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, request_1.mewMsPost('/cache/get', { key: "conf" }, service, isDebugMode)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        }); }; },
        /**
         * 设置缓存
         * @param service
         */
        cacheSet: function (service) { return function (params) { return tslib_1.__awaiter(_this, void 0, void 0, function () {
            return tslib_1.__generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, request_1.mewMsPost('/cache/set', params, service, isDebugMode)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        }); }; }
    };
}
exports.Common = Common;
