// @mew/request/dist/ms/gc.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Gc = void 0;
var request_1 = require("../request");
function Gc(service, isDebugMode) {
    if (isDebugMode === void 0) { isDebugMode = false; }
    return {
        rent: new request_1.Request('/gc/rent', service, isDebugMode),
        type: new request_1.Request('/gc/type', service, isDebugMode),
        step_price: new request_1.Request('/gc/step_price', service, isDebugMode),
        shopping_cart: new request_1.Request('/gc/shopping_cart', service, isDebugMode),
        // class
        rfid: new request_1.Request('/gc/rfid', service, isDebugMode),
        goods: new request_1.Request('/gc/goods', service, isDebugMode),
        location: new request_1.Request('/gc/location', service, isDebugMode),
        garbage: new request_1.Request('/gc/garbage', service, isDebugMode),
        part: new request_1.Request('/gc/part', service, isDebugMode),
        // function
        getVersion: function () { return request_1.mewRequest('/gc/select/version', {}, 'POST', service, isDebugMode); }
    };
}
exports.Gc = Gc;
