// @mew/request/dist/ms/lc.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Lc = void 0;
var tslib_1 = require("tslib");
var request_1 = require("../request");
function Lc(service, isDebugMode) {
    var _this = this;
    if (isDebugMode === void 0) { isDebugMode = false; }
    return {
        log_clean: new request_1.Request('/lc/log_clean', service, isDebugMode),
        log_rent: new request_1.Request('/lc/log_rent', service, isDebugMode),
        log_modification_integral: new request_1.Request('/lc/log_modification_integral', service, isDebugMode),
        email_template: new request_1.Request('/lc/email_template', service, isDebugMode),
        log_measurement: new request_1.Request('/lc/log_measurement', service, isDebugMode),
        log_scrap: new request_1.Request('/lc/log_scrap', service, isDebugMode),
        log_test: new request_1.Request('/lc/log_test', service, isDebugMode),
        log_process: new request_1.Request('/lc/log_process', service, isDebugMode),
        // class
        log: new request_1.Request('/lc/log', service, isDebugMode),
        log_feedback: new request_1.Request('/lc/log_feedback', service, isDebugMode),
        log_alarm: new request_1.Request('/lc/log_alarm', service, isDebugMode),
        log_user_login: new request_1.Request('/lc/log_user_login', service, isDebugMode),
        log_shelf: new request_1.Request('/lc/log_shelf', service, isDebugMode),
        log_charge: new request_1.Request('/lc/log_charge', service, isDebugMode),
        log_apply: new request_1.Request('/lc/log_apply', service, isDebugMode),
        log_lock: new request_1.Request('/lc/log_lock', service, isDebugMode),
        log_epc_goods: new request_1.Request('/lc/log_device_login', service, isDebugMode),
        log_device_login: new request_1.Request('/lc/Request', service, isDebugMode),
        log_purchase: new request_1.Request('/lc/log_purchase', service, isDebugMode),
        log_purchase_details: new request_1.Request('/lc/log_purchase_details', service, isDebugMode),
        log_device_transfer: new request_1.Request('/lc/log_device_transfer', service, isDebugMode),
        log_overhaul: new request_1.Request('/lc/log_overhaul', service, isDebugMode),
        log_periodical_examination: new request_1.Request('/lc/log_periodical_examination', service, isDebugMode),
        log_patrol_inspection: new request_1.Request('/lc/log_patrol_inspection', service, isDebugMode),
        log_pre_inspection: new request_1.Request('/lc/log_pre_inspection', service, isDebugMode),
        log_verification: new request_1.Request('/lc/log_pre_inspection', service, isDebugMode),
        log_sdxx_file: new request_1.Request('/lc/log_sdxx_file', service, isDebugMode),
        log_IO: new request_1.Request('/lc/log_io', service, isDebugMode),
        log_goods: new request_1.Request('/lc/log_goods', service, isDebugMode),
        log_rfid_shelf: new request_1.Request('/lc/log_rfid_shelf', service, isDebugMode),
        log_stock: new request_1.Request('/lc/log_stock', service, isDebugMode),
        log_meter: new request_1.Request('/lc/log_meter', service, isDebugMode),
        log_receive: new request_1.Request('/lc/log_receive', service, isDebugMode),
        log_garbage: new request_1.Request('/lc/log_garbage', service, isDebugMode),
        log_garbage_details: new request_1.Request('/lc/log_garbage_details', service, isDebugMode),
        // function
        getVersion: function () { return request_1.mewRequest('/lc/select/version', {}, 'POST', service, isDebugMode); },
        /**
         * 工具借还记录接口
         * @param params
         */
        toolStatusChange: function (params) { return tslib_1.__awaiter(_this, void 0, void 0, function () {
            var res;
            return tslib_1.__generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, request_1.mewRequest('/lc/log/toolStatus/change', params, 'POST', service, isDebugMode)];
                    case 1:
                        res = _a.sent();
                        console.log(165, res);
                        return [2 /*return*/, res];
                }
            });
        }); },
        /**
         * 插入多条报修记录
         * @param params
         */
        logFeedbackArrInsert: function (params) { return tslib_1.__awaiter(_this, void 0, void 0, function () {
            var url;
            return tslib_1.__generator(this, function (_a) {
                switch (_a.label) {
                    case 0:
                        url = '/lc/log_feedback/insert/toolArr';
                        return [4 /*yield*/, request_1.mewRequest(url, params, 'POST', service, isDebugMode)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        }); },
        logRentInsertBatch: function (params) { return tslib_1.__awaiter(_this, void 0, void 0, function () {
            var url;
            return tslib_1.__generator(this, function (_a) {
                switch (_a.label) {
                    case 0:
                        url = '/lc/log_rent/insert/batch';
                        return [4 /*yield*/, request_1.mewRequest(url, params, 'POST', service, isDebugMode)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        }); },
        logInsertBatch: function (params) { return tslib_1.__awaiter(_this, void 0, void 0, function () {
            return tslib_1.__generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, request_1.mewRequest('/lc/log/insert/batch', params, 'POST', service, isDebugMode)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        }); },
        logFeedbackGoods: function (params) { return tslib_1.__awaiter(_this, void 0, void 0, function () {
            return tslib_1.__generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, request_1.mewRequest('/lc/log_feedback/select/goods', params, 'POST', service, isDebugMode)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        }); },
        sendEmail: function (json) { return request_1.mewRequest('/lc/email_template/sendEmail', json, 'POST', service, isDebugMode); },
        getStepTotalPrice: function (json) { return request_1.mewRequest('/lc/log_rent/getStepTotalPrice', json, 'POST', service, isDebugMode); },
        getTopTenUsageTimes: function (json) { return request_1.mewRequest('/lc/log/getTopTenUsageTimes', json, 'POST', service, isDebugMode); },
    };
}
exports.Lc = Lc;
