// @mew/request/dist/ms/mc.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getAuth = exports.Mc = exports.AppGetDingTalkUserInfo = exports.RoleName = void 0;
var tslib_1 = require("tslib");
var request_1 = require("../request");
// interface
// TODO 此处如果新增，需要修改 initDepartmentList 切记
var RoleName;
(function (RoleName) {
    RoleName["unauthorized"] = "unauthorized";
    RoleName["company"] = "company";
    RoleName["company2"] = "company2";
    RoleName["admin"] = "admin";
    RoleName["super"] = "super";
    RoleName["gtadmin"] = "gtadmin";
    RoleName["gtdepart"] = "gtdepart";
    RoleName["gtdepart1"] = "gtdepart1";
})(RoleName = exports.RoleName || (exports.RoleName = {}));
var AppGetDingTalkUserInfo = /** @class */ (function () {
    function AppGetDingTalkUserInfo() {
    }
    return AppGetDingTalkUserInfo;
}());
exports.AppGetDingTalkUserInfo = AppGetDingTalkUserInfo;
function Mc(service, isDebugMode) {
    var _this = this;
    if (isDebugMode === void 0) { isDebugMode = false; }
    return {
        role: new request_1.Request('/mc/role', service, isDebugMode),
        // class
        manager: new request_1.Request('/mc/manager', service, isDebugMode),
        access: new request_1.Request('/mc/access', service, isDebugMode),
        role_access: new request_1.Request('/mc/role_access', service, isDebugMode),
        app: new request_1.Request('/mc/app', service, isDebugMode),
        // function
        getAuth: function (json) { return request_1.mewRequest('/auth', json, 'POST', service); },
        getVersion: function () { return request_1.mewRequest('/mc/select/version', {}, 'POST', service, isDebugMode); },
        getAccessToken: function (appid) { return tslib_1.__awaiter(_this, void 0, void 0, function () {
            return tslib_1.__generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, request_1.mewRequest('/mc/app/getAccessToken', { appid: appid }, 'POST', service, isDebugMode)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        }); },
        // getCorpAccessToken: async (json:GetAccessTokenProperty): Promise<{ code, msg, data: AccessTokenData }> => {
        //     return await mewRequest( '/mc/app/getCorpAccessToken', json,'POST', service,isDebugMode)
        // },
        /**
         * 废弃，使用 getAppUserInfo
         * @param json
         */
        getDingTalkUserInfo: function (json) { return tslib_1.__awaiter(_this, void 0, void 0, function () {
            return tslib_1.__generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, request_1.mewRequest('/mc/app/getDingTalkUserInfo', json, 'POST', service, isDebugMode)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        }); },
        getAppUserInfo: function (json) { return tslib_1.__awaiter(_this, void 0, void 0, function () {
            return tslib_1.__generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, request_1.mewRequest('/mc/app/getUserInfo', json, 'POST', service, isDebugMode)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        }); },
        getUserPhoneNo: function (json) { return tslib_1.__awaiter(_this, void 0, void 0, function () {
            return tslib_1.__generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, request_1.mewRequest('/mc/app/getUserPhoneNo', json, 'POST', service, isDebugMode)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        }); },
        sendDingTalkMessage: function (json) { return tslib_1.__awaiter(_this, void 0, void 0, function () {
            return tslib_1.__generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, request_1.mewRequest('/mc/app/sendDingTalkMessage', json, 'POST', service, isDebugMode)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        }); },
        sendWeiXinMessage: function (json) { return tslib_1.__awaiter(_this, void 0, void 0, function () {
            return tslib_1.__generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, request_1.mewRequest('/mc/app/sendWxMessage', json, 'POST', service, isDebugMode)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        }); },
        getAccessByRole: function (roleId) {
            return tslib_1.__awaiter(this, void 0, void 0, function () {
                var res, arr, accessIds, arr2, results, arr3;
                var _this = this;
                return tslib_1.__generator(this, function (_a) {
                    switch (_a.label) {
                        case 0: return [4 /*yield*/, this.role_access.select({ roleId: roleId })];
                        case 1:
                            res = _a.sent();
                            console.log(res);
                            arr = res.data;
                            accessIds = [];
                            arr2 = [];
                            if (arr.length > 0) {
                                arr.forEach(function (item) {
                                    accessIds.push(item.accessId);
                                    arr2.push(_this.access.select({ id: item.accessId, type: 1 }));
                                });
                            }
                            return [4 /*yield*/, Promise.all(arr2)];
                        case 2:
                            results = _a.sent();
                            arr3 = [];
                            results.forEach(function (element) {
                                if (element.data.length === 1) {
                                    element.data[0].children = element.data[0].children.filter(function (item) {
                                        return accessIds.indexOf(item.id) >= 0;
                                    });
                                    if (element.data[0].children.length > 0) {
                                        var _loop_1 = function (i) {
                                            var children = element.data[0].children[i].children;
                                            var children2 = [];
                                            if (children.length > 0) {
                                                children.forEach(function (item) {
                                                    for (var j = 0; j < arr.length; j++) {
                                                        if (item.id == arr[j].accessId && roleId == arr[j].roleId) {
                                                            children2.push(item);
                                                        }
                                                    }
                                                });
                                                element.data[0].children[i].children = children2;
                                            }
                                        };
                                        for (var i = 0; i < element.data[0].children.length; i++) {
                                            _loop_1(i);
                                        }
                                    }
                                    arr3.push(element.data[0]);
                                }
                            });
                            // console.log(arr3)
                            return [2 /*return*/, arr3];
                    }
                });
            });
        },
        // 批量修改角色权限
        updateRoleAccessBatch: function (json) { return tslib_1.__awaiter(_this, void 0, void 0, function () {
            var res;
            return tslib_1.__generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, request_1.mewMsPost("/mc/role_access/update/batch", json, service, isDebugMode)];
                    case 1:
                        res = _a.sent();
                        return [2 /*return*/, res];
                }
            });
        }); },
    };
}
exports.Mc = Mc;
function getAuth(json, service) {
    return request_1.mewRequest('/auth', json, 'POST', service);
}
exports.getAuth = getAuth;
