// @mew/request/dist/weixin/weixin.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.WeiXin = void 0;
var tslib_1 = require("tslib");
var mc_1 = require("../ms/mc");
var request_1 = require("../request");
var WeiXin = /** @class */ (function () {
    function WeiXin(appId, service) {
        this._access_token = '';
        this._appId = appId;
        this._service = service;
        this.init();
    }
    WeiXin.prototype.init = function () {
        return tslib_1.__awaiter(this, void 0, void 0, function () {
            var mc, res;
            return tslib_1.__generator(this, function (_a) {
                switch (_a.label) {
                    case 0:
                        mc = mc_1.Mc(this._service);
                        return [4 /*yield*/, mc.getAccessToken(this._appId)];
                    case 1:
                        res = _a.sent();
                        console.log(18, res);
                        this._access_token = res.data.accessToken;
                        return [2 /*return*/];
                }
            });
        });
    };
    WeiXin.prototype.getMsAccessToken = function () {
        return tslib_1.__awaiter(this, void 0, void 0, function () {
            var json, res;
            return tslib_1.__generator(this, function (_a) {
                switch (_a.label) {
                    case 0:
                        json = { username: 'admin', password: '111111' };
                        return [4 /*yield*/, request_1.mewRequest('/auth', json, 'POST', this._service)];
                    case 1:
                        res = _a.sent();
                        return [2 /*return*/, res.access_token];
                }
            });
        });
    };
    /**
     * 仅支持在微信小程序内调用才有效
     */
    WeiXin.prototype.getUserInfo = function (code) {
        return tslib_1.__awaiter(this, void 0, void 0, function () {
            var mc;
            return tslib_1.__generator(this, function (_a) {
                switch (_a.label) {
                    case 0:
                        mc = mc_1.Mc(this._service);
                        return [4 /*yield*/, mc.getAppUserInfo({ appId: this._appId, code: code })];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    WeiXin.prototype.sendTemplateMessage = function (touser, data) {
        return tslib_1.__awaiter(this, void 0, void 0, function () {
            var mc, json;
            return tslib_1.__generator(this, function (_a) {
                switch (_a.label) {
                    case 0:
                        mc = mc_1.Mc(this._service);
                        json = {
                            appId: this._appId,
                            template_id: 'JhO4leaUMd_TeECaM2-n1iFZ8RGSKK9Ww7hA5YJBlqQ',
                            touser: touser,
                            page: 'index',
                            form_id: 'FORMID',
                            emphasis_keyword: 'keyword1.DATA',
                            data: data
                        };
                        return [4 /*yield*/, mc.sendWeiXinMessage(json)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    return WeiXin;
}());
exports.WeiXin = WeiXin;
