// @mew/request/dist/db/db.js
// // import * as fse from 'fs-extra'
// const fs = require('fs-extra')
//
// export interface DbJob {
//     job: {
//         setting: {
//             speed: {
//                 channel: number
//             },
//             errorLimit?: {
//                 record: number
//             }
//         },
//         content: ContentItem[]
//     }
// }
//
// export interface DbReader {
//     name: string,
//     parameter: {
//         username: string,
//         password: string,
//         where: string,
//         column: string[],
//         splitPk: string,
//         connection: [
//             {
//                 table: string[],
//                 jdbcUrl: string[]
//             }
//         ]
//     }
// }
//
// export interface DbWriter {
//     name: string,
//     parameter: {
//         writeMode: string,
//         username: string,
//         password: string,
//         column: string[],
//         session: string[],
//         preSql: string[],
//         connection: [
//             {
//                 jdbcUrl: string,
//                 table: string[],
//             }
//         ]
//     }
// }
//
// export interface ContentItem {
//     reader: DbReader,
//     writer: DbWriter
// }
//
// export function getDbJobs(content:ContentItem[]):DbJob {
//     return {
//         job: {
//             setting: {
//                 speed: {
//                     channel: 1
//                 }
//             },
//             content
//         }
//
//     }
// }
//
// export function getDbJob(readerProperty:GetDbReaderProperty, writerProperty:GetDbWriterProperty):DbJob {
//     return {
//         job: {
//             setting: {
//                 speed: {
//                     channel: 3
//                 },
//                 "errorLimit": {
//                     "record": 5000
//                 }
//             },
//             content: [
//                 {
//                     reader: getDbReader(readerProperty),
//                     writer: getDbWriter(writerProperty)
//                 }
//             ]
//         }
//     }
// }
//
// export interface GetDbReaderProperty {
//     host:string,
//     port:number,
//     username:string,
//     password:string,
//     dbName:string,
//     tableName:string,
//     column:string[],
//     where?:string,
//     splitPk?:string
// }
//
// export function getDbReader(readerProperty:GetDbReaderProperty):DbReader {
//     const { host, port, username, password, dbName, tableName, column, where, splitPk } = readerProperty
//     return {
//         name: 'mysqlreader',
//         parameter: {
//             username,
//             password,
//             where: where ?? '',
//             column,
//             splitPk: splitPk ?? 'id',
//             connection: [
//                 {
//                     table: [tableName],
//                     jdbcUrl: [`jdbc:mysql://${host}:${port}/${dbName}?useUnicode=true&characterEncoding=UTF-8`]
//                 }
//             ],
//         }
//     }
// }
//
// export interface GetDbWriterProperty {
//     host:string,
//     port:number,
//     username:string,
//     password:string,
//     dbName:string,
//     tableName:string,
//     column:string[],
//     writeMode: 'insert' | 'replace' | 'update'
//     preSql?:string[],
//     session?:string[]
// }
//
// export function getDbWriter(writerProperty:GetDbWriterProperty):DbWriter {
//     const { host, port, username, password, dbName, tableName, column, writeMode, preSql, session } = writerProperty
//     return {
//         name: 'mysqlwriter',
//         parameter: {
//             writeMode,
//             username,
//             password,
//             column,
//             session: session ?? [],
//             preSql: preSql ?? [],
//             connection: [
//                 {
//                     jdbcUrl: `jdbc:mysql://${host}:${port}/${dbName}?useUnicode=true&characterEncoding=UTF-8`,
//                     table: [tableName]
//                 }
//             ],
//         }
//     }
// }
//
// export function writeToFile(file: string,object:any) {
//     fs.writeJsonSync(file, object,{spaces:'\t'})
// }
